tests/test_alert_actions.py

```diff
@@ -0,0 +1,242 @@
+        }
+        assert_successful_summary(connector, expected_summary)
+
+    def test_handle_action_when_search_alerts_calls_search(self, mock_py42_client):
+        param = {
+            "username": "gilbert.lequiche@finance.biz",
+            "start_date": "2031-01-01",
+            "end_date": "2031-06-01",
+            "alert_state": "OPEN",
+        }
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        mock_py42_client.alerts.search.assert_called_once()
+
+    def test_handle_action_when_search_alerts_and_all_params_missing_sets_error_message(
+        self, mock_py42_client
+    ):
+        param = {}
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
+
+    def test_handle_action_when_search_alerts_and_date_range_incomplete_sets_error_message(
+        self, mock_py42_client
+    ):
+        param = {"start_date": "2031-01-01"}
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
+
+    def test_handle_action_when_search_alerts_and_date_format_invalid_sets_error_message(
+        self, mock_py42_client
+    ):
+        param = {
+            "start_date": "2031-31-12",
+            "end_date": "2032-31-01",
+        }
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
+
+    def test_handle_action_when_search_alerts_adds_response_items_to_data(
+        self, mock_py42_with_search_alerts
+    ):
+        param = {"username": "barney.frankenberry@chocula.com"}
+        connector = _create_search_alerts_connector(mock_py42_with_search_alerts)
+        connector.handle_action(param)
+        assert_successful_single_data(connector, _MOCK_SEARCH_ALERTS_LIST_RESPONSE)
+
+    def test_handle_action_when_search_alerts_adds_summary(
+        self, mock_py42_with_search_alerts
+    ):
+        param = {"username": "barney.frankenberry@chocula.com"}
+        connector = _create_search_alerts_connector(mock_py42_with_search_alerts)
+        connector.handle_action(param)
+        assert_successful_summary(
+            connector, {"total_count": _MOCK_SEARCH_ALERTS_LIST_RESPONSE["totalCount"]}
+        )
+
+    def test_handle_action_when_search_alerts_and_no_date_range_provided_defaults_to_last_30_days(
+        self, mock_py42_client
+    ):
+        param = {"username": "barney.frankenberry@chocula.com"}
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        actual_query = mock_py42_client.alerts.search.call_args[0][0]
+        # This is not a great assertion...
+        assert len(actual_query._filter_group_list) == 2
+
+    def test_handle_action_when_set_alert_state_calls_update_state(
```

Comment:
for tests like this, can you include an additional `assert_success`? This is a pattern I tried to follow with all the other tests. Its valuable to know that in addition to the function being called with the expected params, its return status was set correctly.